src/fXmlRpc/Parser/XmlChecker.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Copyright (C) 2012-2015
+ * Lars Strojny, InterNations GmbH <lars.strojny@internations.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+namespace fXmlRpc\Parser;
+
+use fXmlRpc\Exception\ParserException;
+
+/**
+ * Class XmlChecker to check is correct XML
+ * @author Piotr Olaszewski <piotroo89@gmail.com>
+ */
+class XmlChecker
+{
+    /**
+     * @param string $toCheck
+     * @throws ParserException
+     */
+    public static function isValid($toCheck)
+    {
+        libxml_use_internal_errors(true);
+
+        $xml = simplexml_load_string($toCheck);
```

Comment:
For `XmlReaderParser` this should work:
```php
$xml->setParserProperty(XMLReader::VALIDATE, true);
var_dump($xml->isValid());
```

The more I think about this option, the more I feel we should add a configuration option to the parsers to enable validation with the default being true to reduce confusion.